### DISCUSSION TITLE
Please help - zigbee2mqtt on Docker stopped working

### DISCUSSION THREAD

**Discussion by chgaertner** [Koenkk/Z-Stack-firmware#472] [Q&A]
Dear all,

**tl;dr: I have two zigbee sticks (Slaesh, Sonoff), both work with ZHA but do not (Sonoff) and stopped to (Slaesh) work with zigbee2mqtt.**

I have been using zigbee2mqtt (with Slaesh's USB stick) successfully for several years with a load of Aqara sensors and some OSRAM and Ikea plugs and switches and Hue lights. Everything is running on docker containers (zigbee2mqtt, HomeAssistant and of course an mqtt server). Lately I upgraded all containers and had some problems with MariaDB (which I use as database system for HomeAssistant) no longer beeing available for ARM/v7. So I decided to take the big step and switch to Raspbian OS 64 bit on a Raspberry Pi 4 (4 GB). I upgraded all containers to their latest version and started zigbee2mqtt and HomeAssistant. Everything went fine for some days. Two days ago I had to plug out and in the Zigbee USB devices during the Raspberry Pi running and from this moment on, zigbee2mqtt failed to be able to connect to the Zigbee stick. This is the log:

_Using '/app/data' as data directory
Zigbee2MQTT:info  2023-08-18 17:41:26: Logging to console and directory: 'data/log/2023-08-18.17-41-26' filename: log.txt
Zigbee2MQTT:info  2023-08-18 17:41:26: Starting Zigbee2MQTT version 1.32.2 (commit #1ec1e57)
Zigbee2MQTT:info  2023-08-18 17:41:26: Starting zigbee-herdsman (0.17.2)
Zigbee2MQTT:error 2023-08-18 17:41:47: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2023-08-18 17:41:47: Failed to start zigbee
Zigbee2MQTT:error 2023-08-18 17:41:47: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
Zigbee2MQTT:error 2023-08-18 17:41:47: Exiting...
Zigbee2MQTT:error 2023-08-18 17:41:47: Error: Failed to connect to the adapter (Error: SRSP - SYS - ping after 6000ms)
    at ZStackAdapter.start (/app/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/zStackAdapter.ts:103:27)
    at Controller.start (/app/node_modules/zigbee-herdsman/src/controller/controller.ts:132:29)
    at Zigbee.start (/app/lib/zigbee.ts:59:27)
    at Controller.start (/app/lib/controller.ts:101:27)
    at start (/app/index.js:107:5)_

I assumed that I have somehow destroyed the Zigbee stick and ordered a SONOFF Zigbee 3.0 USB Dongle (P version). Sadly I had to discover that the brand new stick causes the same error. I have read through the help page mentioned in the log message and did everything which applied to my setup without any success.

Additionally I found out that *both* sticks do perfectly work when they are connected directly to the HomeAssistant container and they can discover Zigbee devices and receive the status correctly. Only zigbee2mqtt is not able to detect the sticks.

I also have flashed the newest coordinator firmware to both sticks according to the zigbee2mqtt homepage. I use the same method to attach the sticks to the docker containers (zigbee2mqtt and - not at the same time! - HomeAssistant):

- /dev/serial/by-id/usb-Silicon_Labs_slae.sh_cc2652rb_stick_-_slaesh_s_iot_stuff_00_13_4A_00_33_99_42_27-if00-port0:/dev/ttyUSB0
... or ...
- /dev/serial/by-id/usb-Silicon_Labs_Sonoff_Zigbee_3.0_USB_Dongle_Plus_0001-if00-port0:/dev/ttyUSB0

I also tried the newest developer version of zigbee2mqtt and an old one 1.29.x because somewhere I read this as a suggestion.

I am clueless and helpless and have no idea what I might be missing. Does anyone of you have an idea? Thank you!

Greetings, Christian

**Comment by chgaertner** [comment]
PS: I also checked from within the zigbee2mqtt docker container:
_test -w /dev/ttyUSB0 && echo success || echo failure_
This command results in "success". So the zigbee stick is definitely visible within the docker container.

**Comment by chgaertner** [comment]
I'm am copying this thread to the zigbee2mqtt discussion forum because I think, the firmware section is not the correct one. Since the devices work with ZHA I suppose it is not a firmware issue.
